xplique/attributions/base.py

```diff
@@ -3,15 +3,42 @@
 """
 
 from abc import ABC, abstractmethod
+import inspect
 import warnings
 
 import tensorflow as tf
 import numpy as np
 
 from ..types import Callable, Dict, Tuple, Union, Optional
-from ..commons import find_layer, tensor_sanitize, batch_predictions_one_hot, \
-                      predictions_one_hot, batch_predictions_one_hot_callable, \
-                      predictions_one_hot_callable
+from ..commons import (find_layer, tensor_sanitize, predictions_operator,
+                      operator_batching, get_gradient_of_operator,
+                      batch_predictions, gradients_predictions, batch_gradients_predictions,
+                      batch_predictions_one_hot_callable, predictions_one_hot_callable,
+                      no_gradients_available, raise_invalid_operator)
+
+def check_operator(operator: Callable):
+    """
+    Check if the operator is valid g(f, x, y) -> tf.Tensor
+    and raise an exception and return true if so.
+
+    Parameters
+    ----------
+    operator
+        Operator to check
+
+    Returns
+    -------
+    is_valid
+        True if the operator is valid, False otherwise.
+    """
+    is_callable = isinstance(operator, Callable)
+    args = inspect.getfullargspec(operator).args
+
+    # we allow operator with optional arguments, but the first 3 must be present
+    is_valid = is_callable and len(args) >= 3
+
+    if not is_valid:
+        raise_invalid_operator()
```

Comment:
This function should not be in `attributions/base.py`, but in the operator file, i.e. `commons/operator.py`.

Reply:
You're right (same suggestion @Agustin-Picard), I have added a better typing.
Also, we discussed with @lucashervier and @AntoninPoche but we plan to dedicate a page on the documentation (replace the callable page) to dive into this `operator` API.
Last but not least, I want to create few notebooks to showcase the possibilities of this framework.
If you have other ideas, don't hesitate ;)